Gems/ROS2/Code/Source/Camera/CameraSensor.h

```diff
@@ -0,0 +1,117 @@
+    //! It creates dedicated rendering pipeline for each camera
+    class CameraSensor
+    {
+    public:
+        //! Initializes rendering pipeline for the camera sensor
+        //! @param cameraSensorDescription - camera sensor description used to create camera pipeline
+        CameraSensor(const CameraSensorDescription& cameraSensorDescription);
+
+        //! Deinitializes rendering pipeline for the camera sensor
+        virtual ~CameraSensor();
+
+        //! Function publish Image Message frame from rendering pipeline
+        //! @param publisher - ROS2 publisher to publish image in future
+        //! @param header - header with filled message information (frame, timestamp, seq)
+        //! @param cameraPose - current camera pose from which the rendering should take place
+        void publishMassage(
+            std::shared_ptr<rclcpp::Publisher<sensor_msgs::msg::Image>> publisher,
+            const AZ::Transform& cameraPose,
+            const std_msgs::msg::Header& header);
+
+        //! Function to get camera sensor description
+        [[nodiscard]] const CameraSensorDescription& GetCameraSensorDescription() const;
+
+    private:
+        //! Function requesting frame from rendering pipeline
+        //! @param cameraPose - current camera pose from which the rendering should take place
+        //! @param callback - callback function object that will be called when capture is ready
+        //!                   it's argument is readback structure containing, among other thins, captured image
+        void RequestFrame(
+            const AZ::Transform& cameraPose, AZStd::function<void(const AZ::RPI::AttachmentReadback::ReadbackResult& result)> callback);
+
+        CameraSensorDescription m_cameraSensorDescription;
+        AZStd::vector<AZStd::string> m_passHierarchy;
+        AZ::RPI::RenderPipelinePtr m_pipeline;
+        AZ::RPI::ViewPtr m_view;
+        AZ::RPI::Scene* m_scene = nullptr;
+        const AZ::Transform kAtomToRos{ AZ::Transform::CreateFromQuaternion(
+            AZ::Quaternion::CreateFromMatrix3x3(AZ::Matrix3x3::CreateFromRows({ 1, 0, 0 }, { 0, -1, 0 }, { 0, 0, -1 }))) };
+        virtual AZStd::string getPipelineTemplateName() = 0;
+        virtual AZStd::string getPipelineTypeName() = 0;
+
+    protected:
+        //! Read and setup Atom Passes
+        void setupPasses();
+    };
+
+    class CameraDepthSensor : public CameraSensor
+    {
+    public:
+        CameraDepthSensor(const CameraSensorDescription& cameraSensorDescription);
+
+    private:
+        virtual AZStd::string getPipelineTemplateName() override;
+        virtual AZStd::string getPipelineTypeName() override;
+    };
+
+    class CameraColorSensor : public CameraSensor
+    {
+    public:
+        CameraColorSensor(const CameraSensorDescription& cameraSensorDescription);
+
+    private:
+        virtual AZStd::string getPipelineTemplateName() override;
+        virtual AZStd::string getPipelineTypeName() override;
```

Comment:
1. These functions names should be UpperCamelCase per the coding standard: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#function-names
2. Be marked `const` per the coding standard: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#constness
3. The `virtual` keyword should be removed when the `override` keyword is used per the coding standard: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#override